packages/extension/src/background/index.ts

```diff
@@ -53,6 +54,20 @@ const sendBootData = async (req, sender) => {
 };
 
 async function handleMessages(message, sender) {
+  const permissions = await browser.permissions.contains({
+    origins: ['*://*/*'],
+  });
+  if (permissions) {
+    await browser.contentScripts.register({
```

Comment:
Can't we leverage the created hook here?

Reply:
Not actually sure if this is used anywhere. I think this has a similar code to the one in the hook. Let me check to remove this quickly and see how it behaves.

Reply:
Oh please don't remove this!
The background worker needs this to spin up the first time the user has given permissions as they technically two different instances.